nav2_dwb_controller/dwb_critics/src/goal_align.cpp

```diff
@@ -46,7 +46,11 @@ void GoalAlignCritic::onInit()
 {
   GoalDistCritic::onInit();
   stop_on_failure_ = false;
-  forward_point_distance_ = nav_2d_utils::searchAndGetParam(nh_, "forward_point_distance", 0.325);
+  forward_point_distance_ =
+    nav_2d_utils::searchAndGetParam(nh_, "forward_point_distance", 0.325);
+  double xy_goal_align_tolerance =
+    nav_2d_utils::searchAndGetParam(nh_, "xy_goal_align_tolerance", 1.0);
```

Comment:
From a parameter side, I think 1.0m is too close to be just now applying this critic. If you're 1 meter out on a large acerkman style robot, that's too close to now have to think about how to deal with your approach to get in the intended orientation. I'd make it something like 5-10m.

I'm not sure what forward point distance does off hand to give any recommended updates to that.

Reply:
As I understand it, forward point distance defines a point a bit ahead of the base_link (0.325m in this case).  From the comments:
```
  // we want the robot nose to be drawn to its final position
  // (before robot turns towards goal orientation), not the end of the
  // path for the robot center. Choosing the final position after
  // turning towards goal orientation causes instability when the
  // robot needs to make a 180 degree turn at the end
```

I have to say, though, I don't fully understand how it works yet.

Reply:
I suppose that all makes sense to me, that would help a little bit with longer robots but given David wrote this for Locus, I'm surprised that was an issue for them and their circular-diffdrive base. With that explanation and what i read, it seems that this is some value in front of the robot to plan against so that when you're aligning to goal you're going to glide into the waypoint with the right orientation rather than planning to have the center of the robot eventually hit the right orientation. Its an interesting work around for not having the global planner be state-based (i.e. that planner cant tell kinematic constraints on turning radius or ability for each given cells, like doing sampling based state validity checking).

Reply:
If you want your defaults to align with the turtlebot3 you could make that smaller, like 0.10, but certainly no larger than whats there today.